### ISSUE TITLE
Event/QuakeML: Enable using multiple tags with the same name

### ISSUE THREAD

**Issue by megies** [obspy/obspy#1976]
Currently with our implementation for custom tags in custom non-QuakeML namespaces in QuakeML through `extra` attribute does not make it possible to attach multiple tags with the same name (since we use a dictionary as underlying structure). This should somehow be changed.. being backwards compatible could get tricky though.

**Comment by krischer** [comment]
There are a bunch of `multidict` implementations out there which would solve this in a fully backwards compatible manner. We could just roll our own if we don't want another dependency or vendor one of the simpler implementations.

**Comment by megies** [comment]
Good idea.

**Comment by markcwill** [comment]
Second this being implemented or vendored, but I wondered how this is implemented in other libs that have similar specs, like HTTP URL query params in `requests`, which allows a `dict`. Looks like `requests` just does it during serialization by checking each key's value type for `list` or iterable, and then iterating at write time.

```python
>>> r = requests.get('http://localhost', params={'repeat': ['one', 'two']})
# ...will get serialized as
'http://localhost?repeat=one&repeat=two'
```
I think this could be implemented in the `_extras` serialization/deserialization by just checking for list and/or pulling everything out into a list and then collapsing anything with len==1? That way you could still use the standard dict (or is it `AttribDict`?) from the user perspective... pretty sure this is what `xmltodict` does too.

**Comment by megies** [comment]
Yeah, I guess both approaches would work in similar ways, with leaving/allowing a plain dict for single instances and multidict or list for multiple instances.

**Comment by krischer** [comment]
:+1: for using lists! Definitely has less complexity!

**Comment by nikosT** [comment]
> +1 for using lists! Definitely has less complexity!

It seems that adding custom tags in ```.extra``` variable, doesn't work for attribute that is a List container. For example: adding a 
```
wave=WaveformStreamID()
wave.extra={'weight':0.5, 'namespace':'custom', type:'attribute'}
FocalMechanism.waveform_id[wave]
```
